xbmc/addons/PVRClient.cpp

```diff
@@ -94,61 +94,40 @@ CPVRClient::~CPVRClient(void)
 
 void CPVRClient::OnDisabled()
 {
-  // restart the PVR manager if we're disabling a client
-  if (CPVRManager::GetInstance().IsStarted() && CPVRManager::GetInstance().RestartManagerOnAddonDisabled())
-    CPVRManager::GetInstance().Start(true);
+  CAddon::OnDisabled();
+  CPVRManager::GetInstance().Clients()->UpdateAddons();
```

Comment:
in order to get rid of singletons, shouldn't you ask the ServiceManager for the PVR instance instead? Or is this change something for later?